signer/local/local.go

```diff
@@ -254,7 +257,53 @@ func (s *Signer) Sign(req signer.SignRequest) (cert []byte, err error) {
 		safeTemplate.SerialNumber = serialNumber
 	}
 
-	return s.sign(&safeTemplate, profile)
+	var certTBS = safeTemplate
+	if len(profile.CTLogServers) == 0 {
+		cert, err = s.sign(&certTBS, profile)
+		return
+	}
+
+	// Add a poison extension which prevents validation
+	var oidCTPoison = []int{1, 3, 6, 1, 4, 1, 11129, 2, 4, 3}
+	var poisonExtension = pkix.Extension{Id: oidCTPoison, Critical: true, Value: []byte{0x05, 0x00}}
+	var poisonedPreCert = certTBS
+	poisonedPreCert.ExtraExtensions = append(safeTemplate.ExtraExtensions, poisonExtension)
+	cert, err = s.sign(&poisonedPreCert, profile)
+
+	if err != nil {
+		return nil, cferr.Wrap(cferr.CertificateError, cferr.Unknown, err)
+	}
+
+	derCert, _ := pem.Decode(cert)
+	prechain := []ct.ASN1Cert{derCert.Bytes, s.ca.Raw}
+
+	var sctList []byte
+
+	for _, server := range profile.CTLogServers {
+		var ctclient = client.New(server)
+		var resp, err = ctclient.AddPreChain(prechain)
+		if err != nil {
+			return nil, cferr.Wrap(cferr.CTError, cferr.Unknown, err)
+		}
+		log.Info(resp)
+		serializedSCT, err := ct.SerializeSCT(*resp)
+		if err != nil {
+			return nil, err
+		}
+		sctList = append(sctList, serializedSCT...)
+	}
+
+	var oidSCTList = []int{1, 3, 6, 1, 4, 1, 11129, 2, 4, 2}
+	var SCTListExtension = pkix.Extension{Id: oidSCTList, Critical: false, Value: sctList}
+	certTBS.ExtraExtensions = append(certTBS.ExtraExtensions, SCTListExtension)
+
+	cert, err = s.sign(&certTBS, profile)
```

Comment:
If the signature fails, we need to wrap the error in a cferr before returning it.